src/hooks/directoryHooks/useGetAllDirectoryPages.ts

```diff
@@ -0,0 +1,60 @@
+import { useQuery, UseQueryOptions, UseQueryResult } from "react-query"
+
+import { DIR_CONTENT_KEY } from "constants/queryKeys"
+
+import useRedirectHook from "hooks/useRedirectHook"
+
+import * as DirectoryService from "services/DirectoryService/index"
+
+import { isAxiosError } from "utils/axios"
+import { useErrorToast } from "utils/toasts"
+
+import { DirectoryParams } from "types/folders"
+import { DEFAULT_RETRY_MSG } from "utils"
+
+const getDirectoryPages = (params: DirectoryParams): Promise<string[]> => {
+  const { siteName, collectionName } = params
+  const collectionParams = {
+    siteName,
+    collectionName,
+  }
+  return DirectoryService.getCollection(collectionParams).then((data) => {
+    return data
+      .map((item) => {
+        if (item.type === "file") {
+          return item.name
+        }
+        return item.children.map((child) => `${item.name}/${child}`)
+      })
+      .flatMap((item) => item)
+  })
+}
+
+export const useGetAllDirectoryPages = (
+  params: DirectoryParams,
+  queryOptions?: Omit<UseQueryOptions<string[]>, "queryFn" | "queryKey">
+): UseQueryResult<string[]> => {
```

Comment:
This hook is called by all `EditPages`, including unlinked and resource pages, which lack a `collectionName` (required by `DirectoryParams`). I think this currently doesn't affect those pages though, since this hook's primary function seems to be for retrieving data to build a page's third nav in the preview, so we can probably avoid calling them? Currently users always see an error popup when loading unlinked/resource pages

Reply:
Ah I missed that error toast when testing. Fixed in 71b89a5e!